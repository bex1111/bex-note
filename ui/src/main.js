import { createApp } from 'vue'
import App from './App.vue'
import PrimeVue from 'primevue/config';
import Button from 'primevue/button';
import Select from 'primevue/select';
import Toolbar from 'primevue/toolbar';
import Card from 'primevue/card';
import InputText from 'primevue/inputtext';
import FloatLabel from 'primevue/floatlabel';
import Fluid from 'primevue/fluid';
import Dialog from 'primevue/dialog';
import Aura from '@primeuix/themes/aura';
import 'primeicons/primeicons.css'
import { createPinia } from 'pinia'
import {useAuthorizationStore} from "./store/authorization";
import './style.css'

const app=createApp(App)

const pinia = createPinia()
app.use(pinia)
export const tokenStore = useAuthorizationStore()

app.use(PrimeVue,{
    theme: {
        preset: Aura
    }
});
app.component('prime-button', Button);
app.component('prime-select', Select);
app.component('prime-toolbar', Toolbar);
app.component('prime-card', Card);
app.component('prime-input', InputText);
app.component('prime-float-label', FloatLabel);
app.component('prime-fluid', Fluid);
app.component('prime-dialog', Dialog);

app.mount('#app')
