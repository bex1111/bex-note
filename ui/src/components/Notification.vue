<script setup>
import { useToast } from "primevue/usetoast";
import {notificationStore} from "../main";

const toast = useToast();

notificationStore.$subscribe((_, state) => {
  toast.add({ severity: state.type, summary: state.message, life: 5000 });
})

</script>

<template>
  <prime-toast/>
</template>

<style scoped>

</style>