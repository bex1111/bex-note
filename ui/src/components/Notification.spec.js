import {mount} from '@vue/test-utils';
import Notification from './Notification.vue';
import {useToast} from "primevue/usetoast";
import {notificationStore} from "../main";

vi.mock('../main', () => ({
    notificationStore: {
        $subscribe: vi.fn()
    }
}));

vi.mock('primevue/usetoast', () => ({
    useToast: vi.fn()
}));

describe('Notification.vue', () => {


    const createWrapper = () => mount(Notification, {
        global: {
            stubs: {
                'prime-toast': true
            }
        }
    });

    beforeEach(() => {
        vi.clearAllMocks();
    });

    it('matches snapshot', () => {
        const wrapper = createWrapper();
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('subscribes to notification store and calls toast.add when state changes', () => {
        let mockAdd = vi.fn();
        useToast.mockReturnValue({
            add: mockAdd
        });
        createWrapper();

        const subscribeCallback = notificationStore.$subscribe.mock.calls[0][0];

        const mockMutation = {};
        const mockState = {
            type: 'success',
            message: 'Note saved successfully'
        };

        subscribeCallback(mockMutation, mockState);

        expect(useToast).toHaveBeenCalledTimes(1);
        expect(mockAdd).toHaveBeenCalledTimes(1);
        expect(mockAdd).toHaveBeenCalledWith({
            severity: 'success',
            summary: 'Note saved successfully',
            life: 5000
        });
    });
});
