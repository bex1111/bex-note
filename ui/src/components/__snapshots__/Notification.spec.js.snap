// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Notification.vue > matches snapshot 1`] = `"<prime-toast-stub></prime-toast-stub>"`;
